builtin/providers/ultradns/resource_ultradns_dirpool.go

```diff
@@ -0,0 +1,612 @@
+				Required: true,
+				// 0-255 char
+			},
+			"rdata": &schema.Schema{
+				Type:     schema.TypeSet,
+				Set:      hashRdatas,
+				Required: true,
+				// Valid: len(rdataInfo) == len(rdata)
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						// Required
+						"host": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"all_non_configured": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+						"geo_info": &schema.Schema{
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"is_account_level": &schema.Schema{
+										Type:     schema.TypeBool,
+										Optional: true,
+										Default:  false,
+									},
+									"codes": &schema.Schema{
+										Type:     schema.TypeSet,
+										Optional: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+										Set:      schema.HashString,
+									},
+								},
+							},
+						},
+						"ip_info": &schema.Schema{
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"is_account_level": &schema.Schema{
+										Type:     schema.TypeBool,
+										Optional: true,
+										Default:  false,
+									},
+									"ips": &schema.Schema{
+										Type:     schema.TypeSet,
+										Optional: true,
+										Set:      hashIPInfoIPs,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"start": &schema.Schema{
+													Type:     schema.TypeString,
+													Optional: true,
+													// ConflictsWith: []string{"cidr", "address"},
+												},
+												"end": &schema.Schema{
+													Type:     schema.TypeString,
+													Optional: true,
+													// ConflictsWith: []string{"cidr", "address"},
+												},
+												"cidr": &schema.Schema{
+													Type:     schema.TypeString,
+													Optional: true,
+													// ConflictsWith: []string{"start", "end", "address"},
+												},
+												"address": &schema.Schema{
+													Type:     schema.TypeString,
+													Optional: true,
+													// ConflictsWith: []string{"start", "end", "cidr"},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			// Optional
+			"ttl": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  3600,
+			},
+			"conflict_resolve": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "GEO",
+				// Valid: "GEO", "IP"
```

Comment:
if there are 2 valid types, it would be worth checking to make sure we only accept those with a validateFunc